### DISCUSSION TITLE
Can't install Homebrew on m1 Mac

### DISCUSSION THREAD

**Discussion by mrekh** [Homebrew/discussions#622] [Everyday usage]
Hi. 
When i paste installation Homebrew command; I get this error:
`==> Downloading and installing Homebrew...
HEAD is now at 10a5434ef Merge pull request #10429 from MikeMcQuaid/yamllint
error: Not a valid ref: refs/remotes/origin/master
fatal: ambiguous argument 'refs/remotes/origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'`
How can I fix this?
And when I want to close the terminal window, i see this: 
`Closing this window will terminate the running processes: bash (4), git (3), git-remote-https.`

**Comment by arwyn0612** [comment]
Having same problem, the computer has been stuck on same message since yesterday.

**Comment by gromgit** [comment]
Do the following:

1. Open a new Terminal window.
2. See if the `brew` script is linked properly: `ls -lL /opt/homebrew/bin/brew`. If you get an error in this step, install Homebrew again.
3. Run the following commands. If you get an error at any stage, post all the output you get:
```
eval "$(/opt/homebrew/bin/brew shellenv)"
brew config
rm -fr $(brew --repo homebrew/core)
brew tap homebrew/core
```

**Reply by lexcraw4d** [reply]
You are my hero! #

**Reply by carlocab** [reply]
If `brew doctor` says `Your system is ready to brew.` then you're all set. If it doesn't, it'll tell you how to fix any problems you might have. Please do what it suggests.

**Reply by daksh3160** [reply]
thanks :)

**Reply by anotherdia** [reply]
Thanks 💯 you save me

**Reply by thayerst2012** [reply]
Been looking for solves and finally this one worked! Thank you!!
